### PR TITLE
refactor(store): protocol code refactoring and api reorganization

### PR DESCRIPTION
**Pull request by LNSD**
The present PR forms part of a series of refactoring PRs that aim to decouple the retention policy from the message store interface completely.

* Expose retention policy execution method and report messages count metric in waku store protocol API.
* Simplify and reorganize the `WakuStore` protocol initialization.
* Improve `wakunode`'s `resume()` procedure error reporting.
* Use only one store instance in waku store protocol implementation. Drop `isSqliteOnly` flag from store protocol.
* Added `DualMessageStore`: in-memory store + persistent store replicating the original behaviour implemented in the waku store protocol.
* Drop waku store protocol `persist_messages` flag. If no store is provided (`nil`) to the protocol, messages won't be persisted.
* Expose two new waku store protocol APIs: report message count metrics and execute retention policy to be actioned externally (from `wakunode`).
* Update tests accordingly.

This is part of the groundwork necessary for #1155.

### CODE REVIEW

**Review by rymnc — APPROVED** [review]
LGTM! 😁

File: waku/v2/protocol/waku_store/protocol.nim [hunk 1]
```diff
@@ -147,8 +146,10 @@ proc findMessages(w: WakuStore, query: HistoryQuery): HistoryResponse {.gcsafe.}
     error: HistoryResponseError.NONE
   )
 
-proc init*(ws: WakuStore, capacity = StoreDefaultCapacity) =
-
+proc initProtocolHandler*(ws: WakuStore) =
+  # TODO: If this line is commented/removed, the code does not compile! WTF!?
```
Comment (jm-clius) [inline waku/v2/protocol/waku_store/protocol.nim:150]: 🤔

Comment (LNSD) [reply waku/v2/protocol/waku_store/protocol.nim:150]: Sorry... My astonishment notes leaked 🤦🏼 

But it is true:

image

And the import is in place:

```nim
  ../../utils/pagination,
```

Comment (LNSD) [reply waku/v2/protocol/waku_store/protocol.nim]: I have solved it by re-exporting the `pagination` module in the `waku_store_queue` module. These are the kind of problems that I don't like 😓 

That's why I work towards a good and easy to follow module graph 🔀

File: waku/v2/node/storage/message/waku_store_queue.nim [hunk 2]
```diff
@@ -11,10 +11,12 @@ import
   ../../../utils/time,
   ./message_store
 
+export pagination
```
Comment (LNSD) [inline waku/v2/node/storage/message/waku_store_queue.nim:14]: 🔧 🤯

File: waku/v2/node/storage/message/dual_message_store.nim [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+method deleteMessagesOlderThanTimestamp*(s: DualMessageStore, ts: Timestamp): MessageStoreResult[void] =
+  # NOTE: Current in-memory store deletes messages as they are inserted. This method fails with a "not implemented" error
+  # ?s.inmemory.deleteMessagesOlderThanTimestamp(ts)
+  ?s.persistent.deleteMessagesOlderThanTimestamp(ts)
+  ok()
+
+method deleteOldestMessagesNotWithinLimit*(s: DualMessageStore, limit: int): MessageStoreResult[void] =
+  # NOTE: Current in-memory store deletes messages as they are inserted. This method fails with a "not implemented" error
+  # ?s.inmemory.deleteOldestMessagesNotWithinLimit(limit)
+  ?s.persistent.deleteOldestMessagesNotWithinLimit(limit)
```
Comment (jm-clius) [inline waku/v2/node/storage/message/dual_message_store.nim:88]: Nice! Amazing how clean this becomes with a properly defined interface.

File: waku/v2/node/storage/message/waku_store_queue.nim (hunk 2 above)
Comment (jm-clius) [reply waku/v2/node/storage/message/waku_store_queue.nim:14]: 🧙‍♂️

**Review by jm-clius — APPROVED** [review]
Yes! Good work. :)

**Pull request merged**